Extension/src/Utility/Filesystem/find.ts

```diff
@@ -0,0 +1,223 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+/* eslint-disable @typescript-eslint/unified-signatures */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+import { Dirent, constants } from 'fs';
+import { readdir, stat } from 'fs/promises';
+import { basename, extname, sep } from 'path';
+import { isWindows } from '../../constants';
+import { accumulator, foreach } from '../Async/iterators';
+import { ManualPromise } from '../Async/manualPromise';
+import { returns } from '../Async/returns';
+import { is } from '../System/guards';
+import { File, Folder, normalize } from './filepath';
+
+interface FolderWithChildren extends Folder {
+    children?: Map<string, File | FolderWithChildren>;
+}
+
+const cache = new Map<string, File | FolderWithChildren | Promise<FolderWithChildren | undefined>>();
+
+/**
+ * This reads a directory and returns a map of the files and folders in it
+ * It is quite tolerant of errors and reentrancy, so if multiple callers are trying to get the same results
+ * it will only do the work once per directory
+ *
+ * @param fullPath the full path of the folder to read
+ * @param executableExtensions  a set of file extensions that are considered executable on windows

```

Comment:
windows->Windows (also there's 2 spaces after the executableExtensions, which is not consistent with the other @param).

Reply:
fixed